sdk/include/sdk/DataPointValues.h

```diff
@@ -0,0 +1,256 @@
+}
+template <> inline DataPointValue::Type getValueType<std::vector<int32_t>>() {
+    return DataPointValue::Type::INT32_ARRAY;
+}
+template <> inline DataPointValue::Type getValueType<int64_t>() {
+    return DataPointValue::Type::INT64;
+}
+template <> inline DataPointValue::Type getValueType<std::vector<int64_t>>() {
+    return DataPointValue::Type::INT64_ARRAY;
+}
+template <> inline DataPointValue::Type getValueType<uint32_t>() {
+    return DataPointValue::Type::UINT32;
+}
+template <> inline DataPointValue::Type getValueType<std::vector<uint32_t>>() {
+    return DataPointValue::Type::UINT32_ARRAY;
+}
+template <> inline DataPointValue::Type getValueType<uint64_t>() {
+    return DataPointValue::Type::UINT64;
+}
+template <> inline DataPointValue::Type getValueType<std::vector<uint64_t>>() {
+    return DataPointValue::Type::UINT64_ARRAY;
+}
+template <> inline DataPointValue::Type getValueType<float>() {
+    return DataPointValue::Type::FLOAT;
+}
+template <> inline DataPointValue::Type getValueType<std::vector<float>>() {
+    return DataPointValue::Type::FLOAT_ARRAY;
+}
+template <> inline DataPointValue::Type getValueType<double>() {
+    return DataPointValue::Type::DOUBLE;
+}
+template <> inline DataPointValue::Type getValueType<std::vector<double>>() {
+    return DataPointValue::Type::DOUBLE_ARRAY;
+}
+template <> inline DataPointValue::Type getValueType<std::string>() {
+    return DataPointValue::Type::STRING;
+}
+template <> inline DataPointValue::Type getValueType<std::vector<std::string>>() {
+    return DataPointValue::Type::STRING_ARRAY;
+}
+
+template <typename T> class TypedDataPointValue : public DataPointValue {
+public:
+    TypedDataPointValue()
+        : DataPointValue(Type::INVALID, "", Timestamp{}){};
+
+    TypedDataPointValue(const std::string& path, T value, Timestamp timestamp = Timestamp{})
+        : DataPointValue(getValueType<T>(), path, std::forward<decltype(timestamp)>(timestamp))
+        , m_value(std::move(value)) {}
+
+    TypedDataPointValue(const std::string& path, DataPointFailure failure,
+                        Timestamp timestamp = Timestamp{})
+        : DataPointValue(getValueType<T>(), path, std::forward<decltype(timestamp)>(timestamp),
+                         failure)
+        , m_value{} {}
+
+    [[nodiscard]] const T& value() const {
+        if (!isValid()) {
+            throw InvalidValueException(
+                fmt::format("'{}' has no valid value: {}!", getPath(), toString(getFailure())));
+        }
+        return m_value;
+    }
+
+    bool operator==(const TypedDataPointValue& other) const {
+        return DataPointValue::operator==(other) && m_value == other.m_value;
+    }
+
+private:
+    T m_value;
+};
+
+using DataPointMap_t = std::map<std::string, std::shared_ptr<DataPointValue>>;
+
+/**
+ * @brief Result of an operation which returns multiple data points.
+ *        Provides typed access to obtained data points.
+ *
+ */
+class DataPointValues final {
```

Comment:
What about
```suggestion
class DataPointValueMap final {
```
I think just adding a `s` will lead to confusions.

Reply:
Class is just used in query (get/subscribe) case. Renaming to DataPointReply to be in sync with Python implementation